### PR TITLE
resubmission, challenges completed

### PR DESCRIPTION
**Pull request by jennerdulce**
# Submitting Week 1 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        const vowelTester1 = "learn"
+        const vowelTester2 = "academy"
+        const vowelTester3 = "challenges"
+        const vowelTester4 = "ytrxz"
+        expect(findFirstVowel(vowelTester1)).toEqual(1)
+        expect(findFirstVowel(vowelTester2)).toEqual(0)
+        expect(findFirstVowel(vowelTester3)).toEqual(2)
+        expect(findFirstVowel(vowelTester4)).toEqual("No vowel found")
+    })
+})
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:166]: Great work expanding the test scenarios!

File: code-challenges.test.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        if( stringToArray[i] === "a" ||
+            stringToArray[i] === "e" ||
+            stringToArray[i] === "i" ||
+            stringToArray[i] === "o" ||
+            stringToArray[i] === "u"){
+                return i
+            }
+    }
+    return "No vowel found"
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:194]: Awesome work!

File: coding-concepts.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+var myCodingSkills = {
+  languages: ["JavaScript", "Ruby"],
+  frameworks: ["React", "Ruby on Rails"],
+  databases: "PostgreSQL",
+  versionControl: "GitHub"
+}
+console.log(myCodingSkills.languages[0])
+
+// a) Your answer: I think the output will be "JavaScript". There are key / value pairs within an object also known as properties. We are able to access properties of an object by using what is called "dot notation" which looks like this `objectName.property`. Accessing the languages property will let us access its value which is an array that contains the strings "JavaScript" and "Ruby". Since the value of myCodingSkills.languages is an array. We are able to then use "index notation" to get the value at a specific index of an array. In this case, we are using the index of 0. The value at this index is the string "JavaScript".
+// b) Verify and explain: My answer was correct, the output was "JavaScript". We accessed a property of an object called languages. The value of this property "languages" was an array containing strings where we then used index notation to access a value at a specified index.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:57]: Expertly articulated!

File: coding-concepts.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    this.cohort = "Bravo",
+    this.year = 2022
+  }
+}
+var learnStudent = new Learn("George")
+console.log(learnStudent)
+
+// a) Your answer: I think that output of this will return a Learn object with the student property change to george, cohort is Bravo, and year is 2022. A class constuctor is created from lines 62 - 68. There are a few properties created within this constructor that are student, cohort, and year. The constuctor also has a parameter of name which will reflect on the student property. This ultimately makes this constructor dynamic and allows users / devs to enter in a name of their choosing when instantiating a new object. Line 69 declares a new variable named learnStudent and this is assigned a value which is a newly instantiated object that derives from the Learn class. A string with George has also been passed as an argument which changes the value of the student property to George
+// b) Verify and explain: The answer I predicted was correct. Logging the variable learnStudent returned an object that was instantiated from the Learn class.
+// Output: Learn { student: 'George', cohort: 'Bravo', year: 2022 }
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:74]: Well done!

File: interview-questions.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. What is the difference between a parameter and an argument?
+
+  Your answer: Overall, a parameter and an argument refer to the same data. When creating a function, a parameter will be the naming convention that acts as the value that is passed in the function. When invoking the function, data may be passed into a function as an argument. The value of the parameter is based on what is passed into the function as an argument. 
+
+  Researched answer: A parameter is a placeholder variable that is definied INSIDE the the parentheses of the function expression  where it may be leveraged/utilized anywhere within the function logic. An argument is the actual value that is passed to the function when the function is invoked. The parameter is then assigned this value.
+
+  - Reference: [Functions Overview](https://github.com/learn-academy-2022-charlie/Syllabus/blob/main/javascript/functions.md)
```
Comment (WalkerTexas-coder) [inline interview-questions.md:15]: Correct the parameter is a local variable that is scoped to the function.

File: interview-questions.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+
+
+4. What is the difference between a function and a method?
+
+  Your answer: The main difference between functions and methods is that methods are functions that are directly tied to a certain type of object. Methods are similar to functions where they execute a code blcok when invoked. However, methods can only be used to the object that they are bound to.
+
+  Researched answer: A function is called a method when it specifically belongs to an Object
+
+  - Reference: [Objects](https://github.com/learn-academy-2022-charlie/Syllabus/blob/main/javascript/objects.md)
```
Comment (WalkerTexas-coder) [inline interview-questions.md:50]: Great work!

**Review by WalkerTexas-coder — APPROVED** [review]
Jenner this assessment is amazing. You've done great work on every part of this and I want you to keep it up next week!

File: interview-questions.md (hunk 5 above)
Comment (jennerdulce) [reply interview-questions.md:15]: Ahhh okay yes, I thought about referring a parameter as a variable but was unsure if that was the correct terminology. Thank you!

**Comment by jennerdulce** [comment]
Thank you!

**Pull request merged**